File: doc-examples/mongo-example-records-java/src/main/java/example/Book.java

package example;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.DateCreated;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import org.bson.types.ObjectId;

import java.util.Date;

// tag::book[]
@MappedEntity
record Book(
        @Id @GeneratedValue ObjectId id,
        @DateCreated @Nullable Date dateCreated,
        String title,
        int pages) {

    Book(String title, int paged) {
        this(null, null, title, paged);
    }

}
// end::book[]

File: doc-examples/mongo-example-kotlin/src/main/kotlin/example/Book.kt

package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import org.bson.types.ObjectId

// tag::book[]
@MappedEntity
data class Book(@field:Id
                @GeneratedValue
                var id: ObjectId,
                var title: String,
                var pages: Int = 0)
// end::book[]